### PR TITLE
feat(*): rework

### PR DESCRIPTION
**Pull request by ToasterSticks**
Same as #13 but I dont want to rebase.

### CODE REVIEW

File: src/utils/highlightMentions.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	switch (type) {
+		case "user": {
+			return isMentioned && id === message.client.user!.id ? bold(blue(match)) : blue(match);
+		}
+		case "role": {
+			const role = message.guild!.roles.cache.get(id);
+
+			if (!role) return match;
+
+			const roleColor = role.color ? role.hexColor : "#d3d3d3";
```
Comment (itsUrcute) [inline src/utils/highlightMentions.ts:20]:
```suggestion
const roleColor = role.hexColor ?? "#d3d3d3";
```

File: src/utils/startEventHandler.ts [hunk 2]
```diff
@@ -1,11 +1,11 @@
+import type { Client, ClientEvents } from "discord.js";
```
Comment (itsUrcute) [inline src/utils/startEventHandler.ts:1]: type imports after normal imports please

**Review by itsUrcute — CHANGES_REQUESTED** [review]

File: src/utils/highlightMentions.ts (hunk 1 above)
Comment (ToasterSticks) [reply src/utils/highlightMentions.ts:19]: Color is used as its a number. Roles with no color will return 0, while hexColor still returns a non falsey string.

Comment (itsUrcute) [reply src/utils/highlightMentions.ts:19]: alright

File: src/commands/channel.ts [hunk 3]
```diff
@@ -1,52 +1,51 @@
...
-- {{prefix}}channel [Server Name] [channel-name]
-- {{prefix}}channel [channel-id]
+			- {{prefix}}channel [Server Name] [channel-name]
+			- {{prefix}}channel [channel-id]
 		`
 	},
-	async execute(input, client: Client) {
-		let channel;
+	async execute(input, client) {
+		let channel: Channel | undefined;
```
Comment (itsUrcute) [inline src/commands/channel.ts:16]:
```suggestion
		let channel;
```

**Review by itsUrcute — CHANGES_REQUESTED** [review]

**Review by itsUrcute — APPROVED** [review]

**Pull request merged**